### PR TITLE
Bring scene editor back 

### PR DESCRIPTION
**Pull request by ragner**
* GVRViewSceneObject: Fix calc of default quad for android renderable views
* GVRViewSceneObject: Rename onStartDraw() -> onStartRendering() event
* GVRViewSceneObject: Add waitFor() method so you can wait for the instance be ready
with renderData built.
* GVRViewSceneOBject: Unregister android view from layout hierarchy when the object is removed from the scene.
* scene editor back

 Known issues that I need help:
* Button "Remove from Scene" at editor dialog is crashing the application. There is something wrong with SceneSerializer.

* Cursor3d looks not working properly. Touch event is always launching a view dialog even if the touch is out of 3d models. Cursor3d is not working with my gear vr oculus just at develop mode touching on screen.
 

GearVRf-DCO-1.0-Signed-off-by: Ragner Magalhaes <ragner.n@samsung.com>

### CODE REVIEW

File: GVRf/Framework/framework/src/main/java/org/gearvrf/scene_objects/GVRViewSceneObject.java [hunk 1]
```diff
@@ -495,42 +592,31 @@ public void onGlobalLayout() {
...
-             */
-            mGVRContext.getActivity().runOnUiThread(new Runnable() {
+            mGVRContext.runOnGlThread(new Runnable() {
                 @Override
                 public void run() {
-                    // To call draw(...) after renderData has been created
-                    setVisibility(VISIBLE);
-
-                    onViewVisible();
+                    mSceneObject.onStartRendering();
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/scene_objects/GVRViewSceneObject.java]: Can you please deliver the callback to the framework thread instead? I would like not to promote access to the rendering thread. In the actual implementation of onStartRendering you can runOnGlThread if it is necessary.

**Comment by liaxim** [comment]
When I try it, as soon as I touch the screen, the view freezes and then the editor crashes with:
```
03-08 11:03:28.118 23044 23044 E AndroidRuntime: FATAL EXCEPTION: main
03-08 11:03:28.118 23044 23044 E AndroidRuntime: Process: org.gearvrf.io.sceneeditor, PID: 23044
03-08 11:03:28.118 23044 23044 E AndroidRuntime: java.lang.UnsupportedOperationException: GVRSVewSceneObject#waitFor() cannot lock UI thread.
03-08 11:03:28.118 23044 23044 E AndroidRuntime: 	at org.gearvrf.scene_objects.GVRViewSceneObject.waitFor(GVRViewSceneObject.java:323)
03-08 11:03:28.118 23044 23044 E AndroidRuntime: 	at org.gearvrf.io.sceneeditor.BaseView.<init>(BaseView.java:40)
03-08 11:03:28.118 23044 23044 E AndroidRuntime: 	at org.gearvrf.io.sceneeditor.FileBrowserView.<init>(FileBrowserView.java:67)
03-08 11:03:28.118 23044 23044 E AndroidRuntime: 	at org.gearvrf.io.sceneeditor.SceneEditorMain$6.onStateChanged(SceneEditorMain.java:252)
03-08 11:03:28.118 23044 23044 E AndroidRuntime: 	at org.gearvrf.io.cursor3d.SelectableBehavior.setState(SelectableBehavior.java:313)
03-08 11:03:28.118 23044 23044 E AndroidRuntime: 	at org.gearvrf.io.cursor3d.SelectableBehavior.setButtonPress(SelectableBehavior.java:257)
03-08 11:03:28.118 23044 23044 E AndroidRuntime: 	at org.gearvrf.io.cursor3d.CursorManager$3.onTouchStart(CursorManager.java:1307)
03-08 11:03:28.118 23044 23044 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
--
03-08 11:03:28.118 23044 23044 E AndroidRuntime: 	at android.os.MessageQueue.next(MessageQueue.java:323)
03-08 11:03:28.118 23044 23044 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:136)
03-08 11:03:28.118 23044 23044 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6776)
03-08 11:03:28.118 23044 23044 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-08 11:03:28.118 23044 23044 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520)
03-08 11:03:28.118 23044 23044 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410)
```

**Comment by ragner** [comment]
I hope it works now @liaxim

**Comment by liaxim** [comment]
The scene editor works now. The demos though need updates. gvr-events crashes with this for me:

```
03-09 07:55:06.308 10759 10759 E AndroidRuntime: FATAL EXCEPTION: main
03-09 07:55:06.308 10759 10759 E AndroidRuntime: Process: org.gearvrf.events, PID: 10759
03-09 07:55:06.308 10759 10759 E AndroidRuntime: java.lang.IllegalStateException: The specified child already has a parent. You must call removeView() on the child's parent first.
03-09 07:55:06.308 10759 10759 E AndroidRuntime: 	at android.view.ViewGroup.addViewInner(ViewGroup.java:4636)
03-09 07:55:06.308 10759 10759 E AndroidRuntime: 	at android.view.ViewGroup.addView(ViewGroup.java:4477)
03-09 07:55:06.308 10759 10759 E AndroidRuntime: 	at android.view.ViewGroup.addView(ViewGroup.java:4417)
03-09 07:55:06.308 10759 10759 E AndroidRuntime: 	at android.view.ViewGroup.addView(ViewGroup.java:4390)
03-09 07:55:06.308 10759 10759 E AndroidRuntime: 	at org.gearvrf.GVRActivity$2.run(GVRActivity.java:557)
03-09 07:55:06.308 10759 10759 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:751)
03-09 07:55:06.308 10759 10759 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:95)
03-09 07:55:06.308 10759 10759 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:154)
```

**Comment by ragner** [comment]
@liaxim done!

**Comment by ragner** [comment]
@liaxim Currently when you  remove a GVRViewSceneObject from the scene its internal Android view keeps register to android view hierarchy. I will fix that issue at some next PR for now I am just turning the View invisible to avoid onDraw() calls.

**Comment by liaxim** [comment]
Ok. gvr-events works now. gvr-sceneobjects fails with:
```
03-12 09:54:43.150 30739 30896 W System.err: android.view.ViewRootImpl$CalledFromWrongThreadException: Only the original thread that created a view hierarchy can touch its views.
03-12 09:54:43.153 30739 30896 W System.err: 	at android.view.ViewRootImpl.checkThread(ViewRootImpl.java:7913)
03-12 09:54:43.153 30739 30896 W System.err: 	at android.view.ViewRootImpl.invalidateChildInParent(ViewRootImpl.java:1373)
03-12 09:54:43.154 30739 30896 W System.err: 	at android.view.ViewGroup.invalidateChild(ViewGroup.java:5449)
03-12 09:54:43.154 30739 30896 W System.err: 	at android.view.View.invalidateInternal(View.java:14825)
03-12 09:54:43.154 30739 30896 W System.err: 	at android.view.View.invalidate(View.java:14789)
03-12 09:54:43.154 30739 30896 W System.err: 	at android.view.View.setFlags(View.java:12618)
03-12 09:54:43.154 30739 30896 W System.err: 	at android.view.View.setVisibility(View.java:8673)
03-12 09:54:43.154 30739 30896 W System.err: 	at org.gearvrf.scene_objects.GVRViewSceneObject.onNewParentObject(GVRViewSceneObject.java:253)
03-12 09:54:43.154 30739 30896 W System.err: 	at org.gearvrf.GVRSceneObject.addChildObject(GVRSceneObject.java:685)
03-12 09:54:43.154 30739 30896 W System.err: 	at org.gearvrf.GVRScene.addSceneObject(GVRScene.java:122)
03-12 09:54:43.154 30739 30896 W System.err: 	at org.gearvrf.sample.sceneobjects.SampleMain.onInit(SampleMain.java:146)
03-12 09:54:43.155 30739 30896 W System.err: 	at java.lang.reflect.Method.invoke(Native Method)
03-12 09:54:43.155 30739 30896 W System.err: 	at org.gearvrf.GVREventManager.invokeMethod(GVREventManager.java:307)
03-12 09:54:43.155 30739 30896 W System.err: 	at org.gearvrf.GVREventManager.sendEventWithMaskParamArray(GVREventManager.java:126)
03-12 09:54:43.155 30739 30896 W System.err: 	at org.gearvrf.GVREventManager.sendEventWithMask(GVREventManager.java:113)
03-12 09:54:43.155 30739 30896 W System.err: 	at org.gearvrf.GVREventManager.sendEvent(GVREventManager.java:108)
03-12 09:54:43.155 30739 30896 W System.err: 	at org.gearvrf.GVRViewManager$1$1.run(GVRViewManager.java:340)
03-12 09:54:43.155 30739 30896 W System.err: 	at android.os.Handler.handleCallback(Handler.java:751)
03-12 09:54:43.155 30739 30896 W System.err: 	at android.os.Handler.dispatchMessage(Handler.java:95)
03-12 09:54:43.155 30739 30896 W System.err: 	at android.os.Looper.loop(Looper.java:154)
03-12 09:54:43.155 30739 30896 W System.err: 	at android.os.HandlerThread.run(HandlerThread.java:61)
```

**Comment by ragner** [comment]
Hi @liaxim,
I have tested my last commit with scene-editor, gvr-renderableviews, gvr-events and gvr-sceneobjects.

**Comment by liaxim** [comment]
Looks good. Planning to merge on Friday.

**Pull request merged**